src/main/java/com/socrata/datasync/validation/GISJobValidity.java

```diff
@@ -0,0 +1,758 @@
+     * @param location a LocationColumn
+     * @param typeMapping a map from dataset column field names to their type
+     * @return true if each component column has a supported type; false if any have an unsupported type
+     */
+    private static JobStatus locationTypesSupported(String locationName,
+                                                    LocationColumn location,
+                                                    Map<String,String> typeMapping) {
+        Map<String,String> componentColumns = location.findComponentColumns();
+
+        for (String component : componentColumns.keySet()){
+            String fieldname = componentColumns.get(component);
+            String[] supportedTypes;
+            if (component.equalsIgnoreCase("zip") || component.equalsIgnoreCase("latitude") || component.equalsIgnoreCase("longitude")) {
+                supportedTypes = new String[]{"text", "number"};
+            } else {
+                supportedTypes = new String[]{"text"};
+            }
+
+            String unsupportedType = getUnsupportedType(fieldname, supportedTypes, typeMapping);
+            if (unsupportedType != null) {
+                JobStatus status = JobStatus.PUBLISH_ERROR;
+                StringBuilder message = new StringBuilder("Unsupported Datatype: " +
+                                                          Utils.capitalizeFirstLetter(component) +
+                                                          " component  '" +
+                                                          locationName + "' is of type '" +
+                                                          unsupportedType + "'; " +
+                                                          component + " components require ");
+
+                if (supportedTypes.length == 1) {
+                    message.append(" a ");
+                }
+
+                message.append("'" + supportedTypes[0] + "'");
+                for (int i=1; i<supportedTypes.length; i++) {
+                    message.append(" or '" + supportedTypes[i] + "'");
+                }
+
+                message.append(" column");
+                if (supportedTypes.length > 1) {
+                    message.append('s');
+                }
+
+                message.append(".\nPlease specify a column that matches a supported type.");
+                status.setMessage(message.toString());
+                return status;
+            }
+        }
+
+        return  JobStatus.VALID;
+    }
+
+    /**
+     * returns null if the component type is okay, otherwise the bad type.
+     */
+    private static String getUnsupportedType(String component,
+                                             String[] supportedTypes,
+                                             Map<String, String> typeMapping) {
+        String componentType = typeMapping.get(component);
+        if (component == null || componentType == null) {
```

Comment:
If component is null, this will have NPE'd by this point if typeMapping does not happen to be a type that allows null keys (most don't)